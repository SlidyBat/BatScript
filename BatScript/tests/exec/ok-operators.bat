print 10 + 5
print 10.0 + 5.1
print 10 - 5
print 10.0 - 5.1
print 10 / 5
print 10 / 7
print 10.0 / 7
print 10 * 5
print 10.0 * 5.1
print 1 << 1
print 4 >> 1
print 5 % 2
print ~(-1)
print 2 ^ 1
print 4 | 1
print 3 & 1
print !true
print !false
print !1
print !0
print 10 < 5
print 10 < 20
print 10 > 5
print 10 > 20
print 10.0 < 5.0
print 10.0 < 20.0
print 10.0 > 5
print 10.0 > 20.0
print 10 <= 10
print 10 <= 5
print 10 <= 20
print 10.0 <= 10.0
print 10.0 <= 5.0
print 10.0 <= 20.0
print 10 >= 10
print 10 >= 5
print 10 >= 20
print 10.0 >= 10.0
print 10.0 >= 5.0
print 10.0 >= 20.0
print 10 == 5
print 10 == 10
print 10 == 20
print 10.0 == 5.0
print 10.0 == 10.0
print 10.0 == 20.0

int a = 5
print a
int b = 10
print b
b += a
print b
b -= a
print b
b /= a
print b
b *= a
print b
b %= a
print b
b |= a
print b
b ^= a
print b
b = 10
b ^= a
print b