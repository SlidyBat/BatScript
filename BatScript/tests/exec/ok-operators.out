15
15.1
5
4.9
2
1
1.42857
50
51
2
2
1
0
3
5
1
0
1
0
1
0
1
1
0
0
1
1
0
1
0
1
1
0
1
1
1
0
1
1
0
0
1
0
0
1
0
5
10
15
10
2
10
0
5
0
15
